frontend/app/services/letters-service.server.ts

```diff
@@ -119,8 +73,78 @@ function createInteropService() {
     });
   }
 
+  /**
+   * @returns array of letters given the userId and clientId with optional sort parameter
+   */
+  async function getLetterInfoByClientId(userId: string, clientId: string, sortOrder: 'asc' | 'desc' = 'desc') {
+    const url = new URL(`${CCT_API_BASE_URI}/cctws/OnDemand/api/GetDocInfoByClientId`);
+    url.searchParams.set('userid', userId);
+    url.searchParams.set('clientid', clientId);
+    url.searchParams.set('community', CCT_VAULT_COMMUNITY);
+    url.searchParams.set('Exact', 'true');
+
+    const response = await fetch(url);
+
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed to fetch data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const lettersSchema = z.array(
+      z.object({
+        LetterRecordId: z.string().optional(),
+        LetterDate: z.string().optional(),
+        LetterId: z.string().optional(),
+        LetterName: z.string().optional(),
+      }),
+    );
+
+    const letters = lettersSchema.parse(await response.json()).map((letter) => ({
+      id: letter.LetterRecordId,
+      issuedOn: letter.LetterDate,
+      name: letter.LetterName,
+      referenceId: letter.LetterId,
+    }));
+
+    return sort(letters, {
+      order: sortOrder,
+      by: (item) => item.issuedOn ?? 'undefined',
+    });
+  }
+
+  async function getPdf(userId: string, referenceId: string) {
+    const url = new URL(`${CCT_API_BASE_URI}/cctws/OnDemand/api/GetPdfByLetterId`);
+    url.searchParams.set('community', CCT_VAULT_COMMUNITY);
+    url.searchParams.set('id', referenceId);
+    url.searchParams.set('userid', userId);
+
+    const response = await fetch(url);
+
+    if (response.ok || response.status === 404) {
+      return response;
+    }
+
+    log.error('%j', {
+      message: 'Failed to fetch data',
+      status: response.status,
+      statusText: response.statusText,
+      url: url,
+      handlersresponseBody: await response.text(),
+    });
+
+    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
```

Comment:
Can you reverse the order of checks here?
```
if (!ok) { /* error */ }
return success
```

Reply:
Fixed